### PR TITLE
Feature/20 1 user account enviroment

### PR DESCRIPTION
**Pull request by katarzynasitarz**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,14 +9,28 @@ import { Todolist } from "./components/Todolist";
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: null
+    };
+  }
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged(user => {
+      if (user) {
```
Comment (cytrowski) [inline src/App.js:28]: I think we do not need the `if` statement here. If you log out you probably want to set user back to `null`

File: src/App.js [hunk 2]
```diff
@@ -25,8 +40,18 @@ export class App extends Component {
...
+                  <Route path="/" exact component={Catalog} />
+                ) : (
+                  <Route path="/" exact component={LoginSite} />
+                )}
+
                 <Route path="/add-form" component={AddForm} />
-                <Route path="/" exact component={Catalog} />
                 <Route path="/calendar" component={Calendar} />
                 <Route path="/todolist" component={Todolist} />
                 <Route path="/wishlist" component={Wishlist} />
```
Comment (cytrowski) [inline src/App.js:57]: Those should be put inside the `this.state.user ? .... : ....` expression above as well

File: src/components/Footer.js [hunk 3]
```diff
@@ -1,7 +1,32 @@
...
+    this.state = {
+      user: null
+    };
+  }
+
+  handleLogout = () => {
+    firebase
+      .auth()
+      .signOut()
+      .then((window.location = "/"));
```
Comment (cytrowski) [inline src/components/Footer.js:17]: In `then` you should put a function, eg. `() => { window.location = '/' }`

File: src/components/LoginSite.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    firebase
+      .auth()
+      .signInWithEmailAndPassword(email, password)
+      .then(user => {
+        this.props.history.push("/");
+      })
+      .catch(error => {
+        this.setState({ error });
+      });
+    console.log("zalogowano");
```
Comment (cytrowski) [inline src/components/LoginSite.js:30]: This is true in between lines 24 and 26 - not here :)

File: src/components/LoginSite.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  };
+
+  render() {
+    return (
+      <>
```
Comment (cytrowski) [inline src/components/LoginSite.js:35]: No need for `React.Fragment` - `<>` here :) We have a single `div` inside

File: src/components/PlantsGrid.js [hunk 6]
```diff
@@ -58,7 +58,7 @@ export class PlantsGrid extends Component {
         </div>
         <div className="grid-display">
           {filteredPlants.map(plant => {
-            console.log(plant)
+            console.log(plant);
```
Comment (cytrowski) [inline src/components/PlantsGrid.js:61]: We can get rid of this console.log call

File: src/components/LoginSite.js (hunk 4 above)
Comment (katarzynasitarz) [reply src/components/LoginSite.js:30]: But it works in this placem I checked :D I truly don't know, how to put it between 24 - 26.

**Pull request merged**